src/core/tasks.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import absolute_import
+import csv
+import ftplib
+import os
+
+from celery import task
+from django.conf import settings
+
+from core import models, email
+from services import ServiceHandler, JuraUpdateService
+
+
+def _read_csv(path):
+    """Read a CSV and return its rows, cleaning NUL bytes."""
+    old_csv = open(path, 'rb')
+    data = old_csv.read()
+    old_csv.close()
+    if path != 'bisac.csv':
+        os.remove(path)
+    new_csv = open('new.csv', 'wb')
+    new_csv.write(
+        data.replace('\00', '')
+    )
+    new_csv.close()
+
+    with open('new.csv', 'rb') as f:
+        reader = csv.reader(
+            f.read().splitlines()
+        )
+        return reader
+
+
+@task(name='add-metadata')
+def add_metadata():
+    """Adds book metadata from CSVs on an FTP site.
+
+    Accesses a specified FTP dir, copies CSVs to
+    the local dir then deletes them from FTP, reads
+    the CSV and matches ISBN13s to Rua books and
+    updates their metadata with the matching CSV
+    column values, and sends a notification email.
+    """
+    if settings.FTP_METADATA_FEATURE:
+        service = ServiceHandler(service=JuraUpdateService)
+
+        ftp = ftplib.FTP(getattr(settings, 'FTP_URL', None))
+        ftp.login(
+            getattr(settings, 'FTP_USERNAME', None),
+            getattr(settings, 'FTP_PASSWORD', None)
+        )
+        ftp.cwd(getattr(settings, 'FTP_FOLDER', None))
+        files = []
+
+        try:
+            files = ftp.nlst()
+        except ftplib.error_perm, resp:
+            if str(resp) == '550 No files found':
+                print 'No files in this directory'
+            else:
+                raise
+
+        for f in files:
+            if '.csv' in f:
+                ftp.retrbinary(
+                    'RETR ' + f,
+                    open(f, 'wb').write
+                )
+                ftp.delete(f)
+                ftp.close()
+
+                _csv = _read_csv(f)
+                headers = next(_csv)
+                isbn_index = headers.index('ISBN13')
+                bisac_index = headers.index('Bisac Code(s)')
+                description_index = headers.index('Long Description')
+                data = [row for row in _csv]
```

Comment:
I don't see the point in this when you could use ```_csv``` directly in the for loop thus

```for row in _csv:```

doing ```data = [row for row in _csv]``` seems excessive and makes the code a little harder to read.